### PR TITLE
feat(propagation):  Add models for Action feature settings

### PR DESCRIPTION
**Pull request by samblackk**
This PR adds model extensions to support action feature toggling.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **New Features**
	- Introduced new GraphQL data fetchers for managing document propagation settings.
	- Added queries and mutations for fetching and updating document propagation settings.
	- Enhanced API capabilities by adding new global settings management features.

- **Bug Fixes**
	- Improved error handling for settings retrieval and updates to enhance user experience.

- **Documentation**
	- Updated GraphQL schema documentation to include new fields for document propagation settings.

- **Chores**
	- Standardized JSON formatting for policy definitions to improve readability.

### CODE REVIEW

File: datahub-graphql-core/src/main/resources/app.graphql
Comment (shirshanka) [inline datahub-graphql-core/src/main/resources/app.graphql:552]: can be deleted

**Comment by samblackk** [comment]
@shirshanka Resolved.

**Review by shirshanka — APPROVED** [review]
Approved - merge once CI is green.

**Review by jjoyce0510 — APPROVED** [review]
Would prefer to reuse existing query and mutation fields for global settings, as they already support different types of settings. 

I won't block on it though. For future automations, I'd prefer we extend instead of creating net new.

**Pull request merged**
